src/pytest_inmanta_lsm/plugin.py

```diff
@@ -340,17 +344,17 @@ def get_optional_option(option: StringTestParameter) -> Optional[str]:
     # Configure the client once more, to make sure we can cleanup everything behind us
     remote_orchestrator.setup_config()
 
-    # If --lsm-no-clean is used, leave the orchestrator as it is, with all its
-    # file, otherwise cleanup the project
-    if not remote_orchestrator_no_clean:
-        remote_orchestrator.clear_environment()
-
     # If --lsm-no-halt is used, leave the orchestrator running at the end of the
     # test suite.  Otherwise the environment is halted.
     if not remote_orchestrator_no_halt:
```

Comment:
Should we then always halt the environment when we don't specify `--lsm-no-clean`? (meaning `--lsm-no-halt` can't be used/has not effect without `--lsm-no-clean`)

Reply:
This PR just become a bit more complicates, because it was decided to implement https://github.com/inmanta/inmanta-core/issues/6587. I am going to unassign you from review for now and rework this PR based on the above-mentioned ticket.